packages/editor/src/components/header/style.scss

```diff
@@ -43,6 +43,11 @@
 	flex-grow: 1;
 	display: flex;
 	justify-content: center;
+	// Flex items will, by default, refuse to shrink below a minimum
+	// intrinsic width. In order to shrink this flexbox item, and
+	// subsequently truncate child text, we set an explicit min-width.
+	// See https://dev.w3.org/csswg/css-flexbox/#min-size-auto
+	min-width: 0;
```

Comment:
Borrowed comment from the document bar styles. https://github.com/WordPress/gutenberg/blob/0a4c5c2f38d801eedd0de2b6e54258e687442777/packages/editor/src/components/document-bar/style.scss#L6-L10